### ISSUE TITLE
Field value is always undefined

### ISSUE THREAD

**Issue by afitiskin** [erikras/redux-form#174]
I have simple form, connected to the redux store:
```javascript
@reduxForm({
    fields: ['firstname', 'lastname'],
    form: 'profileEditForm',
})
export default class ProfileEditForm extends Component {
    render() {
        const { fields } = this.props;
        console.log(fields.firstname.value); // always undefined
        /* rendering form ... */
    }
}
```
When I do `console.log(somefield.value)` I always get `undefined`. So that means that all fields are *not controlled*. Is that correct behaviour?

**Comment by erikras** [comment]
I'd need to see more of your `rendering form`. You will need to supply `fields.firstname.onChange` to your input components, either manually or with a restructured props: `<input type="text" {...fields.firstname}/>`

**Comment by afitiskin** [comment]
I do it with restructured props: {...fields.firstname}

**Comment by erikras** [comment]
Hmm. Not sure what to tell you. Simplify your form little by little until you get closer to the [Simple Form Example](http://erikras.github.io/redux-form/#/examples/simple)?

**Comment by afitiskin** [comment]
I've tried to simplify my code as much as possible. So at the moment it looks very close to Simple Form Example: 

In my app.jsx:
```javascript
import React from 'react';
import ReactDOM from 'react-dom';
import { combineReducers, createStore } from 'redux';
import { Provider, connect } from 'react-redux';
import { reduxForm, reducer as formReducer } from 'redux-form';

const reducer = combineReducers({
    form: formReducer,
});
const store = createStore(reducer);

@reduxForm({
    fields: ['firstname', 'lastname'],
    form: 'profileEditForm',
})
class ProfileEditForm extends React.Component {
    render() {
        const { handleSubmit, fields } = this.props;
        console.log('render', fields.firstname.value);

        return (
            <form onSubmit={handleSubmit}>
                <label>First name:</label>
                <input type="text" {...fields.firstname} />
                <label>Last name:</label>
                <input type="text" {...fields.lastname} />
                <button type="submit">Submit</button>
            </form>
        );
    }
}

@connect()
class App extends React.Component {
    submit() {
        console.log('submit');
    }
    render() {
        console.log('render app');        
        return (
            <ProfileEditForm onSubmit={this.submit} />
        )
    }
}

const app = (
    <Provider store={store}>
        <App />
    </Provider>
);

ReactDOM.render(app, document.getElementById('content'));
```

The first thing that I did not find in the documentation is the fact of passing dispatch function to form component. So when I ran this code, in the console I got the error message: 
```
Warning: Failed propType: Required prop `dispatch` was not specified in `ReduxForm(ProfileEditForm)`. Check the render method of `App`.
```

So, to prevent this error, I pass `dispatch` function to `ProfileEditForm` component:
```javascript
    render() {
        console.log('render app');
        return (
            <ProfileEditForm onSubmit={this.submit} dispatch={this.props.dispatch} />
        )
    }
```

Error message gone after that change, but both `console.log` executes only once. Nothing changed when I type into fields.

So, I tried to check if data in store changed when I type. I changed App component to pass form branch of state to the props:
```javascript
@connect(state => ({
    form: state.form
}))
class App extends React.Component {
    submit() {
        console.log('submit');
    }
    render() {
        console.log('render app', this.props.form);
        return (
            <ProfileEditForm {...this.props} onSubmit={this.submit} />
        )
    }
}
```

I was surprised, but my code start works as I want! Each changes in fields immediately executes both render functions. `console.log('render', fields.firstname.value);` logs correct value of first name field.

I tried to go deeper, and only pass necessary data to `App`:
```
@connect(state => ({
    form: state.form.profileEditForm
}))
class App extends React.Component {
    submit() {
        console.log('submit');
    }
    render() {
        console.log('render app', this.props.form);
        return (
            <ProfileEditForm {...this.props} onSubmit={this.submit} />
        )
    }
}
```

And again I was surprised, but this changes in code move me back: render functions executes on each changes, but `console.log('render', fields.firstname.value);` always logs `undefined`. 
Am I doing something wrong?

**Comment by afitiskin** [comment]
My `package.json` file:
```json
{
  "dependencies": {
    "react": "^0.14.0",
    "react-dom": "^0.14.0",
    "react-redux": "^4.0.0",
    "react-router": "^1.0.0-rc1",
    "redux": "^3.0.0",
    "redux-form": "^2.4.5",
    "redux-router": "^1.0.0-beta3",
    "redux-thunk": "^0.1.0",
  },
  "devDependencies": {
    "babel-core": "^5.4.7",
    "babel-eslint": "^3.1.9",
    "babel-loader": "^5.1.2",
    "css-loader": "^0.21.0",
    "eslint": "^1.7.3",
    "eslint-plugin-react": "^2.3.0",
    "node-sass": "^3.3.3",
    "react-hot-loader": "^1.2.7",
    "sass-loader": "^2.0.1",
    "style-loader": "^0.12.3",
    "webpack": "^1.9.6",
    "webpack-dev-server": "^1.8.2"
  }
}
```

**Comment by afitiskin** [comment]
Working code for copy-paste:
```javascript
import React from 'react';
import ReactDOM from 'react-dom';
import { combineReducers, createStore } from 'redux';
import { Provider, connect } from 'react-redux';
import { reduxForm, reducer as formReducer } from 'redux-form';

const reducer = combineReducers({
    form: formReducer,
});
const store = createStore(reducer);

@reduxForm({
    fields: ['firstname', 'lastname'],
    form: 'profileEditForm',
})
class ProfileEditForm extends React.Component {
    render() {
        const { handleSubmit, fields } = this.props;
        console.log('render', fields.firstname.value);

        return (
            <form onSubmit={handleSubmit}>
                <label>First name:</label>
                <input type="text" {...fields.firstname} />
                <label>Last name:</label>
                <input type="text" {...fields.lastname} />
                <button type="submit">Submit</button>
            </form>
        );
    }
}

@connect(state => ({
    form: state.form
}))
class App extends React.Component {
    submit = () => {
        console.log('submit');
    }
    render() {
        console.log('render app', this.props.form);
        return (
            <ProfileEditForm {...this.props} onSubmit={this.submit} />
        )
    }
}

const app = (
    <Provider store={store}>
        <App />
    </Provider>
);

ReactDOM.render(app, document.getElementById('content'));
```

This is only works correct when `{...this.props}` passed to `ProfileEditForm`: `<ProfileEditForm {...this.props} onSubmit={this.submit} />`

**Comment by afitiskin** [comment]
I've checked everything again and I found that redux-form version was 2.4.5 instead of latest one, so all the problems was because of that. I've updated the version of redux-form and removed all hacks from code – everything works as expected! I'm sorry for my stupid mistake. 

P.S. I think it will be good to add installation code in README file as: 
```shell
npm install redux-form@3.0.0-beta-2 --save
```
to prevent same errors :)
